packages/ember-page-layout/src/services/layout.ts

```diff
@@ -0,0 +1,20 @@
+import Service from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+
+export default class Layout extends Service {
+    @tracked sidebarTree: string[] = [];
+    @tracked headerTree: string[] = [];
+    @tracked footerTree: string[] = [];
+
+    get currentSidebar() {
+        return this.sidebarTree[this.sidebarTree.length - 1];
+    }
+
+    get currentHeader() {
+        return this.headerTree[this.headerTree.length - 1];
+    }
+
+    get currentFooter() {
+        return this.footerTree[this.footerTree.length - 1];
+    }
```

Comment:
more of a phase 2 enhancement, but i could see us generalizing this to allow any number of different page layout "trees", for example if a app had a secondary nav or maybe an "aside", right sidebar or something, etc.  Perhaps the consuming app could just pass in an arbitrary string for what "stack"/tree the layout  component should belong in, and the array would get auto-created in the service if it doesn't already exist.